editor/src/core/property-controls/property-controls-local.spec.tsx

```diff
@@ -233,6 +278,117 @@ describe('registered property controls', () => {
       ]
     `)
   })
+  it('updating the control registration removes the build errors', async () => {
+    const renderResult = await renderTestEditorWithModel(
+      project({
+        ['/utopia/components.utopia.js']: `import { Card } from '../src/card'
+        
+        const Components = {
+      '/src/card': {
+        Cart: {
+          component: Card,
+          supportsChildren: false,
+          properties: { },
+          variants: [ ],
+        },
+      },
+    }
+    
+    export default Components
+  `,
+      }),
+      'await-first-dom-report',
+    )
+
+    expect(renderResult.getEditorState().editor.codeEditorErrors).toEqual({
+      buildErrors: {},
+      lintErrors: {
+        '/utopia/components.utopia.js': [
+          {
+            codeSnippet: '',
+            endColumn: null,
+            endLine: null,
+            errorCode: '',
+            fileName: '/utopia/components.utopia.js',
+            message:
+              'Validation failed: Component name (Card) does not match the registration key (Cart)',
+            passTime: null,
+            severity: 'fatal',
+            source: 'eslint',
+            startColumn: null,
+            startLine: null,
+            type: '',
+          },
+        ],
+      },
+    })
+
+    // /src/card is not here
```

Comment:
I still think it would be nicer to expect() that /src/card is not in the array, instead of checking what _is_ in the array. 

if we add a new default builtin test dependency, all of these tests will fail

Reply:
makes sense, refactored the code in https://github.com/concrete-utopia/utopia/pull/5126/commits/eeace749ca3cf3e056d4163d89d2ee483eeb1c58